Test/patterns/PatternMatching.dfy

```diff
@@ -0,0 +1,89 @@
+// RUN: %dafny /compile:3 "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+/*
+*  This file contains a collection of tests for features modified or introduced in PR 458
+*/
+
+
+datatype Alt = A(int) | B(int)
+datatype MyOption = Some(v: Alt) | None
+datatype MyPair = Pair(x:Alt, y:Alt)
+datatype List<T> = Nil | Cons(head: T, tail: List)
+
+// Nested Patterns
+method NestingTest (xs:List<int>) returns (r:int)
+{
+    match xs
+     case Cons(y, Cons(z, zs)) => return z;
+     case Cons(y, Nil) => return y;
+     case Nil => return 0;
```

Comment:
Indent with 2 spaces. Also, line up each `case` with `match` (since this `match` does not use curly braces around the `case`s).